modules/core/include/opencv2/core/quaternion.hpp

```diff
@@ -0,0 +1,1949 @@
+         * For example
+         * ```
+         * quatd q(1,2,3,4);
+         * power(q, 2);
+         *
+         * double angle = CV_PI;
+         * Vec3d axis{0, 0, 1};
+         * Quatd q1(angle, axis); //generate a unit quat by axis and angle
+         * power(q1, 2, true);
+         * ```
+         */
+        template <typename T, typename _T>
+        friend Quat<T> power(const Quat<T> &q, _T x, bool assumeUnit);
+
+        /**
+         * @brief return the value of power function with index \f$x\f$
+         * \f[q^x = ||q||(\cos(x\theta) + \boldsymbol{u}\sin(x\theta)))\f]
+         * @param x index of exponentiation
+         * @param assumeUnit if true, this quaternion should be a unit quaternion and this function will save some computations.
+         *
+         * For example
+         * ```
+         * quatd q(1,2,3,4);
+         * q.power(2);
+         *
+         * double angle = CV_PI;
+         * Vec3d axis{0, 0, 1};
+         * Quatd q1(angle, axis); //generate a unit quat by axis and angle
+         * q1.power(2, true); //true means q is a unit quaternion
+         * ```
+         */
+        template <typename _T>
+        Quat<_Tp> power(_T x, bool assumeUnit=false) const;
+
+        /**
+         * @brief return \f$\sqrt{q}\f$
+         * @param q a quaternion
+         * @param assumeUnit if true, quaternion q should be a unit quaternion and this function will save some computations.
+         *
+         * For example
+         * ```
+         * Quatf q(1,2,3,4);
+         * sqrt(q);
+         *
+         * q = {1,0,0,0};
+         * sqrt(q, true); //true means q is a unit quaternion
+         * ```
+         */
+        template <typename T>
+        friend Quat<T> sqrt(const Quat<T> &q, bool assumeUnit);
+
+        /**
+         * @brief return \f$\sqrt{q}\f$
+         * @param assumeUnit if true, this quaternion should be a unit quaternion and this function will save some computations.
+         *
+         * For example
+         * ```
+         * Quatf q(1,2,3,4);
+         * q.sqrt();
+         *
+         * q = {1,0,0,0};
+         * q.sqrt(true); //true means q is a unit quaternion
+         * ```
+         */
+        Quat<_Tp> sqrt(bool assumeUnit=false) const;
+
+        /**
+         * @brief return the value of power function with quaternion \f$q\f$
+         * \f[p^q = e^{q\ln(p)}\f]
+         * @param p base quaternion of power function
+         * @param q index quaternion of power function
+         * @param assumeUnit if true, quaternon \f$p\f$ should be a unit quaternion and this function will save some computations
+         *
+         * For example
+         * ```
+         * Quatd p(1,2,3,4);
+         * Quatd q(5,6,7,8);
+         * power(p, q);
+         *
+         * p = p.normalize();
+         * power(p, q, true); //true means p is a unit quaternion
+         * ```
+         */
+        template <typename T>
+        friend Quat<T> power(const Quat<T> &p, const Quat<T> &q, bool assumeUnit);
```

Comment:
It can be implemented without friendship, because there is public method  `power`.